distribution/tools/keystore-cli/src/main/java/org/elasticsearch/cli/keystore/AddStringKeyStoreCommand.java

```diff
@@ -53,42 +50,26 @@ protected void executeCommand(Terminal terminal, OptionSet options, Environment
 
         final KeyStoreWrapper keyStore = getKeyStore();
 
-        final Closeable closeable;
-        final CheckedFunction<String, char[], IOException> valueSupplier;
-        if (options.has(stdinOption)) {
-            final Reader stdinReader = terminal.getReader();
-            valueSupplier = s -> {
-                try (CharArrayWriter writer = new CharArrayWriter()) {
-                    int c;
-                    while ((c = stdinReader.read()) != -1) {
-                        if ((char) c == '\r' || (char) c == '\n') {
-                            break;
-                        }
-                        writer.write((char) c);
-                    }
-                    return writer.toCharArray();
-                }
-            };
-            closeable = stdinReader;
-        } else {
-            valueSupplier = s -> terminal.readSecret("Enter value for " + s + ": ");
-            closeable = () -> {};
-        }
+        final CheckedFunction<String, char[], IOException> valueSupplier = s -> {
+            String prompt = "";
+            if (options.has(stdinOption) == false) {
+                prompt = "Enter value for " + s + ": ";
+            }
```

Comment:
This style of supplying a default that is sometimes overridden later defeats Java's error detection for paths where the value is not set.

```suggestion
            String prompt;
            if (options.has(stdinOption)) {
                prompt = "";
            } else {
                prompt = "Enter value for " + s + ": ";
            }
```

Reply:
I applied your suggestion.